network_simulator/device.py

```diff
@@ -0,0 +1,79 @@
+"""This module contains all device definitions.
+
+.. autosummary::
+    Host
+    Router
+"""
+
+# Imports go here.
+# e.g.    from .common import ACK_PACKET_SIZE_BITS
+
+class Device(object):
+    """Representation of a device (Host or Router).
+
+    :param str name: name of this device from the input file.
+    :param str address: IP address of this device.
+    :ivar str name: name of this device from the input file.
+    :ivar str address: IP address of this device.
+    """
+    # TODO(team): Do we need a "name" here?
+    def __init__(self, name=None, address=None):
+        self.name = name
+        self.address = address
+
+    def __eq__(self, other):
+        return self.address == other.address
+
+    def __neq__(self, other):
+        return not __neq__(self, other)
+
+    # TODO(team): More event subclasses to come?
+
+class Host(Device):
+    """Representation of a host.
+    A host sends :class:`Packets <.Packet>` through a :class:`.Link` to a
+    :class:`.Router` or a :class:`.Host`.
+
+    :param dict flows: :class:`Flows <.Flow>` from this :class:`.Host`.
```

Comment:
Include just the :ivar in the documentation, as you did in Router. Otherwise it's repetitive.